File: java10_2darr/untitled/src/p5.java

//find row with max sum
public class p5 {
    public static void main(String[] args) {
        int[][] arr = {{1,2,3,6},{5,61,7,4},{7,8,6,9}};
        int row = 0;
        int max = 0;
        for (int i = 0; i < arr.length; i++) {
            int sum = 0;
            for (int j = 0; j < arr[i].length; j++) {
                if(arr[i][j]>max){
                    sum = sum + arr[i][j];
                }
            }
            if(max < sum){
                max = sum;
                row = i;
            }
        }
        System.out.println(row);// 1
    }
}

File: java19_tree_bst_p2/bst_p2/src/p5.js

//Paths from root with a specified sum gfg practic

/* A Tree node
function Node(key) {
    this.key = key;
    this.left = null;
    this.right = null;
}*/


function op(root, targetSum,domey,i,arr){
  if(root == null){
    return;
  }
  i = i +root.key

  domey.push(root.key);
  if(targetSum == i){
   arr.push([...domey]);
  }
  op(root.left, targetSum,domey,i,arr)
  op(root.right, targetSum,domey,i,arr)
  domey.pop();
}



class Solution {

printPaths(root, targetSum) {
 let domey = [];
 let arr = [];
 let i = 0;
 op(root, targetSum,domey,i,arr);
 return arr;
    }
}